### PR TITLE
[Ability] Implement the secondary effect with Super Luck and Compound Eyes

### PR DESCRIPTION
**Pull request by EvasiveAce**
> [!Warning]
> The PR is now maintained by @DayKev 

## What are the changes?

Added the Gen VIII out of battle effect for Super Luck, and Compound Eyes:
## Why am I doing these changes?

Both Super Luck and Compound Eyes are partially implemented, so I thought I would fix them.

## What did change?

I added the 20% extra enemy held item modifier, so if you give `BonusItemChance` to an ability, it should make enemies 20% more likely to have a held item.

### Screenshots/Videos

Video of the chances of an enemy holding an item with Super Luck/Compound Eyes.

[Video](https://github.com/pagefaultgames/pokerogue/assets/71776311/e7d75b24-b05b-4781-a597-60e86e594346)

Video of the chances of an enemy holding an item without Super Luck/Compound Eyes.

[Video](https://github.com/pagefaultgames/pokerogue/assets/71776311/eb291c48-26cd-4f2a-bf1c-c61de8cad41c)

## How to test the changes?

Give yourself the ability Super Luck/Compound Eyes, and set some console.logs before the chances are set/changed in battle-phase.ts, and after they are set/changed. Check `BonusItemChance` documentation for more information.

## Checklist
- [x] There is no overlap with another PR?
- [x] The PR is self-contained and cannot be split into smaller PRs?
- [x] Have I provided a clear explanation of the changes?
- [x] Have I tested the changes (manually)?
    - [x] Are all unit tests still passing? (`npm run test`)
- [x] Are the changes visual?
  - [x] Have I provided screenshots/videos of the changes?

### CODE REVIEW

**Review by OrangeRed — APPROVED** [review]
Logically this looks fine. What remains is the decision of whether this is behavior is desirable.

**Comment by flx-sta** [comment]
### Are you still working on this PR?
-----
No change since June 3 and open conflicts

**Comment by EvasiveAce** [comment]
I am not currently working on this or #1927. Feel free to fork and fix the conflicts.

**Comment by flx-sta** [comment]
> for now just close it and dont give them an effect at all until a better effect is thought up of
> since giving enemies items is just hurting the player more often then not
- Balance Team

**Comment by DayKev** [comment]
image

File: src/battle-scene.ts [hunk 1]
```diff
@@ -2467,6 +2467,9 @@ export default class BattleScene extends SceneBase {
       const difficultyWaveIndex = this.gameMode.getWaveForDifficulty(this.currentBattle.waveIndex);
       const isFinalBoss = this.gameMode.isWaveFinal(this.currentBattle.waveIndex);
       let chances = Math.ceil(difficultyWaveIndex / 10);
+      if (this.getPlayerPokemon()?.getAbility().hasAttr(BonusItemChanceAbAttr) || this.getPlayerPokemon()?.getPassiveAbility().hasAttr(BonusItemChanceAbAttr)) {
```
Comment (torranx) [inline src/battle-scene.ts:2470]: i think `hasAbility` should cover both active and passive abilities

Comment (innerthunder) [reply src/battle-scene.ts:2470]:
```suggestion
      if (this.getPlayerPokemon()?.hasAbilityWithAttr(BonusItemChanceAbAttr)) {
```
better to use `hasAbilityWithAttr` here

File: src/battle-scene.ts [hunk 2]
```diff
@@ -2478,6 +2478,9 @@ export default class BattleScene extends SceneBase {
       const difficultyWaveIndex = this.gameMode.getWaveForDifficulty(this.currentBattle.waveIndex);
       const isFinalBoss = this.gameMode.isWaveFinal(this.currentBattle.waveIndex);
       let chances = Math.ceil(difficultyWaveIndex / 10);
+      if (this.getPlayerPokemon()?.hasAbilityWithAttr(BonusItemChanceAbAttr)) {
+        chances += Math.floor(chances * 1.2);
```
Comment (flx-sta) [inline src/battle-scene.ts:2482]: @DayKev should this live in a constant in `src/constants.ts`?

maybe move this one too then:
```ts
      if (isFinalBoss) {
        chances = Math.ceil(chances * 2.5);
      }
```

**Review by flx-sta — CHANGES_REQUESTED** [review]
Just 2 minor NIT

File: src/battle-scene.ts (hunk 2 above)
Comment (DayKev) [reply src/battle-scene.ts]: Replaced the magic numbers.

**Review by flx-sta — APPROVED** [review]

**Comment by flx-sta** [comment]
Closed due to rejection by @pagefaultgames/balance-team .
If needed again re-open

**Pull request closed without merging**